aleo-programs/staking_lite/src/main.leo

```diff
@@ -52,13 +58,18 @@ program staking_lite.aleo {
   mapping withdrawals: address => withdrawal_state;
 
   transition initialize(commission_rate: u128, validator_address: address) {
-    assert_eq(self.caller, ADMIN);
+    assert_eq(self.signer, ADMIN);
     assert(commission_rate < PRECISION_UNSIGNED);
 
     return then finalize(commission_rate, validator_address);
   }
 
   finalize initialize(commission_rate: u128, validator_address: address) {
+    // Ensure program is only initialized once
+    let initialized: u8 = is_initalized.get_or_use(0u8, 0u8);
+    assert_eq(initialized, 0u8);
```

Comment:
I added these checks everywhere. It just prevents a lot of potentially undefined behavior.